text/0000-cargo-install.md

```diff
@@ -0,0 +1,271 @@
+cross-compiled binaries to ship to other platforms.
+
+#### Conflicting Crates
+
+Cargo will not namespace the installation directory for crates, so conflicts may
+arise in terms of binary names. For example if crates A and B both provide a
+binary called `foo` they cannot be both installed at once. Cargo will reject
+these situations and recommend that a binary is selected via `--bin` or the
+conflicting crate is uninstalled.
+
+## Managing Installations
+
+If Cargo gives access to installing packages, it should surely provide the
+ability to manage what's installed! The first part of this is just discovering
+what's installed, and this is provided via `cargo install --list`. A more
+interesting aspect is the `cargo install --update` command.
+
+#### Updating Crates
+
+Once a crate is installed new versions can be released or perhaps the build
+configuration wants to be tweaked, so Cargo will provide the ability to update
+crates in-place. By default *something* needs to be specified to the `--update`
+flag, either a specific crate that's been installed or the `--all` flag to
+update all crates. Because multiple crates of the same name can come from
+different sources, the argument to the `--update` flag will be a package id
+specification instead of just the name of a crate.
+
+When updating a crate, it will first attempt to update the source code for the
+crate. For crates.io sources this means that it will download the most recent
+version. For git sources it means the git repo will be updated, but the same
+branch/tag will be used (if original specified when installed). Git sources
+installed via `--rev` won't be updated.
+
+After the source code has been updated, the crate will be rebuilt according to
+the flags specified on the command line. This will override the flags that were
+previously used to install a crate, for example activated features are not
+remembered.
+
+#### Removing Crates
+
+To remove an installed crate, another subcommand will be added to Cargo:
+
+```
+Remove a locally installed crate
+
+Usage:
+    cargo uninstall [options] SPEC
+
+Options:
+    -h, --help              Print this message
+    --bin NAME              Only uninstall the binary NAME
+    --example EXAMPLE       Only uninstall the example EXAMPLE
+    -v, --verbose           Use verbose output
+
+The argument SPEC is a package id specification (see `cargo help pkgid`) to
+specify which crate should be uninstalled. By default all binaries are
+uninstalled for a crate but the `--bin` and `--example` flags can be used to
+only uninstall particular binaries.
+```
+
+Cargo won't remove the source for uninstalled crates, just the binaries that
+were installed by Cargo itself.
+
+## Non-binary artifacts
+
+Cargo will not currently attempt to manage anything other than a binary artifact
+of `cargo build`. For example the following items will not be available to
+installed crates:
+
+* Dynamic native libraries built as part of `cargo build`.
```

Comment:
Now that you mention it, yes, most binaries will be linked fully statically. Except there will surely exist those few odd binaries that link to something dynamically and fail to run at all, no matter what.

Sure, that’s probably not something worth putting a lot of attention to for the first viable release of the functionality.